lib/jekyll/commands/console.rb

```diff
@@ -0,0 +1,115 @@
+      COLORED_BUILD_METHOD_NAMES = BUILD_METHODS.map(&:cyan).join(", ")
+      TIMED_BUILD_METHOD_NAMES = BUILD_METHODS.first(2).map! { |ph| "timed_#{ph}".cyan }.join(", ")
+      private_constant :CONSOLE_DEPS, :BUILD_METHODS, :COLORED_BUILD_METHOD_NAMES,
+                       :TIMED_BUILD_METHOD_NAMES
+
+      class << self
+        # rubocop:disable Metrics/MethodLength
+        def init_with_program(prog)
+          prog.command(:console) do |c|
+            c.syntax "console"
+            c.description <<~DESC.rstrip
+              Start an interactive console preloaded with a site instance.
+
+                  The site instance can be accessed in the console by invoking #{"site".cyan}.
+
+                  This instance responds to additional (console-only) methods that allows
+                  one to instrument the instance for more information. The build methods
+                  #{COLORED_BUILD_METHOD_NAMES} have a `timed_`
+                  equivalent namely #{TIMED_BUILD_METHOD_NAMES}, etc that output the
+                  time taken to run the build method.
+
+                  In additon, the instance also responds to the following console-only methods:
+                    - pre_render  : Outputs time taken to run phases #{"reset".cyan} > #{"read".cyan} > #{"generate".cyan}.
+                    - post_render : Outputs time taken to run phases #{"cleanup".cyan} > #{"write".cyan}.
+
+                  Invoke #{"exit".yellow} or press #{"Ctrl+D".yellow} to exit console.
+            DESC
+            c.alias(:c)
+
+            c.option "config", "--config CONFIG_FILE[,CONFIG_FILE2,...]", Array,
+                     "Custom configuration file(s)"
+            c.option "blank", "--blank", "Preload a bare site"
+            c.option "processed", "--processed", "Preload with a fully built site"
+
+            c.action do |_, options|
+              process(options)
+            end
+          end
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        def process(options)
+          ENV["JEKYLL_ENV"] ||= "console"
+
+          require_relative "console/site"
+          Jekyll::ConsoleSite.new(Jekyll.configuration(options)).tap do |site|
+            log_messages(site)
+            process_site(site, options)
+          end
+
+          start_console
+        end
+
+        private
+
+        def log_messages(site)
+          log "Source:", site.source
+          log "Destination:", site.dest
+          log "Console:", "Bare site initialized."
+          log "", "ProTips:"
+          log "", "* Access the site instance by invoking #{"site".cyan} in the console."
+          log "", "* Invoke #{"exit".yellow} or press #{"Ctrl+D".yellow} to exit console."
+        end
+
+        def start_console
+          print "\nInitializing console "
+
+          60.times do
+            print "."
+            sleep 0.03
+          end
+          log "\n\n"
```

Comment:
What do you think about removing the printing of the 60 dots here? It really don't do anything other than be an artificial delay. Does it have some other purpose?

Reply:
You're right in stating that it doesn't do anything but create an artificial delay.
I intended it to be an animation just to spice things up slightly.
I'll remove it.